### ISSUE TITLE
warning.file.reserved_name_mismatch incorrectly fired for non-label files (e.g. CSV collection inventory files)

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS-Incubator/validate#1580]
## Checked for duplicates
Yes - I've already checked

## 🐛 Describe the bug
`RegisterTargets.checkReservedFilenameMismatch()` fires `warning.file.reserved_name_mismatch` for any file whose basename starts with `collection_` or `bundle_`, regardless of file extension. This incorrectly flags collection inventory CSV files and other non-label files.

**Example:** A collection inventory file named `collection_suda_data_sci_calibrated.csv` triggers:
```
File name 'collection_suda_data_sci_calibrated.csv' uses the reserved 'collection_' pattern
but the product type is not Product_Collection. Reserved file names should only be used for
their intended product types (PDS4 SR 6C.1.3).
```

**Root cause:** `Utility.getTargetType()` calls `TargetExaminer.isTargetCollectionType()`, which only returns `COLLECTION` for XML label files. A `collection_*.csv` file returns `TargetType.FILE`, so the mismatch check fires. The fix is to return early from `checkReservedFilenameMismatch()` if the file does not have a label extension (`.xml` or `.lblx`), since PDS4 SR 6C.1.3 reserved file name rules apply only to label files.

Relevant code: `src/main/java/gov/nasa/pds/tools/validate/rule/RegisterTargets.java` → `checkReservedFilenameMismatch()`

## 🕵️ Expected behavior
`warning.file.reserved_name_mismatch` should only be reported for label files (`.xml`, `.lblx`) whose filename matches a reserved pattern but whose product type does not match. Non-label files (e.g. `.csv` inventory files) with reserved name prefixes should not trigger this warning.

## 📜 To Reproduce
1. Create a PDS4 bundle with a collection whose inventory CSV is named `collection_*.csv`
2. Run: `validate --skip-context-validation --skip-product-validation -R pds4.bundle -t bundle.xml`
3. Observe spurious `warning.file.reserved_name_mismatch` warnings for the CSV file

See test case `github1481` — expects `totalWarnings=1` (`warning.integrity.reference_not_found`) but gets 7 due to extra `warning.file.reserved_name_mismatch` warnings on the CSV inventory file.

## 🖥 Environment Info
- Version of this software: 4.1.0-SNAPSHOT
- Operating System: Linux / macOS

## 📚 Version of Software Used
4.1.0-SNAPSHOT

## 🩺 Test Data / Additional context
Test data: `src/test/resources/github1481/` — contains `data/collection_data.csv` which triggers the false positive.

Also causes false positives for test `github1100` on `collection_suda_data_sci_calibrated.csv`.

## 🦄 Related requirements
- PDS4 Standards Reference 6C.1.3 — Reserved File Names (applies to label files only)
- https://github.com/NASA-PDS/validate/issues/1459

---

**Comment by jordanpadams** [comment]
Skipping I&T rationale: Tested per test cases from https://github.com/NASA-PDS/validate/issues/1459
